xtra-libp2p/src/endpoint.rs

```diff
@@ -59,6 +64,31 @@ pub struct Endpoint {
     inflight_connections: HashSet<PeerId>,
     connection_timeout: Duration,
     subscribers: Subscribers,
+    open_substream_failure_counter: FailureCounter,
+}
+
+#[derive(Debug, Clone)]
+struct FailureCounter {
+    limit: u32,
+    counter: u32,
+}
+
+impl FailureCounter {
+    pub fn new(limit: u32) -> Self {
+        Self { limit, counter: 0 }
+    }
+
+    pub fn increment(&mut self) {
+        self.counter += 1;
+    }
+
+    pub fn limit_reached(&self) -> bool {
+        self.counter < self.limit
```

Comment:
🛠 Looks odd

If `counter=1` and `limit=10` then `limit_reached=true` because `1<10=true`

Is that what you want?

Reply:
Thanks! That was just so wrong 😅

Should have added tests in the first place 😬